vms/evm/predicate/slots.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package predicate
+
+import (
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/types"
+)
+
+type PredicaterExistChecker interface {
+	PredicaterExists(common.Address) bool
+}
```

Comment:
It's not clear what this abstraction is to me... what if we named this something closer to what its usage is in the codebase? It seems like some name reflecting that predicates are a subset of precompiles would make sense here?

```suggestion
type Precompiles interface {
	HasPredicate(address common.Address) bool
}
```